include/parallel/parallel_implementation.h

```diff
@@ -1374,6 +1375,147 @@ inline void Communicator::send (const unsigned int dest_processor_id,
 }
 
 
+
+template <typename T>
+inline void Communicator::send (const unsigned int dest_processor_id,
+                                const std::vector<std::vector<T>> & buf,
+                                const MessageTag & tag) const
+{
+  this->send(dest_processor_id, buf,
+             StandardType<T>((buf.empty() || buf.front().empty()) ?
+                             libmesh_nullptr : &(buf.front().front())), tag);
+}
+
+
+
+template <typename T>
+inline void Communicator::send (const unsigned int dest_processor_id,
+                                const std::vector<std::vector<T>> & buf,
+                                Request & req,
+                                const MessageTag & tag) const
+{
+  this->send(dest_processor_id, buf,
+             StandardType<T>((buf.empty() || buf.front().empty()) ?
+                             libmesh_nullptr : &(buf.front().front())), req, tag);
+}
+
+
+
+template <typename T>
+inline void Communicator::send (const unsigned int dest_processor_id,
+                                const std::vector<std::vector<T>> & buf,
+                                const DataType & type,
+                                const MessageTag & tag) const
+{
+  LOG_SCOPE("send()", "Parallel");
+
+  const std::size_t n_vecs = buf.size();
+
+  // We'll do evil casts to pack metadata+data into the same buffer
+  const std::size_t headersize = (n_vecs+1) * (sizeof(std::size_t)/sizeof(T));
+
+  std::size_t datasize = 0;
+  for (auto & subvec : buf)
+    datasize += subvec.size();
+
+  const std::size_t fullsize = (headersize+datasize)*sizeof(T);
+
+  // Because of those evil casts we don't even want to trigger
+  // constructors/destructors
+  T * tempbuf = (T*)malloc(fullsize);
```

Comment:
Ugh, we're purposely putting malloc/free pairs into the library 😭

Reply:
Yeah, I'm going to take that out.

Reply:
Wouldn't have uploaded first except that Derek needs to use this for something and I had already nearly finished the dumb implementation before realizing what I should have done instead.

Reply:
Oh, I see 👍

Reply:
I am going to leave that PostWaitFreeBuffer class in, because I've known the pain of using a C library that returns heap allocated pointers to C++ code, but I don't intend to use it internally.